docs/src/proposals/multi-writers.md

```diff
@@ -0,0 +1,188 @@
+Similar to the received field, `first_shred_timestamp` can use merge operator
+that includes min operation to correctly capture the `first_shred_timestamp` to
+avoid the case where more than one writers are eligible for updating the
+`first_shred_timestamp` field to avoid potential override in case we want to
+keep this field accurate.
+
+#### [last_index](https://github.com/solana-labs/solana/blob/b57097ef18ab250aeb4ea8d3d111748ab4c41440/ledger/src/blockstore_meta.rs#L26-L27)
+The last_index field doesn't require merge operator to ensure its atomicity as
+there should be only one shred which has its [is_last_in_slot to be true]
+(https://github.com/solana-labs/solana/blob/b57097ef18ab250aeb4ea8d3d111748ab4c41440/ledger/src/blockstore.rs#L3189-L3200).
+
+#### [parent_slot](https://github.com/solana-labs/solana/blob/b57097ef18ab250aeb4ea8d3d111748ab4c41440/ledger/src/blockstore_meta.rs#L28-L29)
+parent_slot is updated only from [std::u64::MAX to a valid parent slot](https://github.com/solana-labs/solana/blob/b57097ef18ab250aeb4ea8d3d111748ab4c41440/ledger/src/blockstore.rs#L3251-L3253).
+A merge operator is needed to ensure we never override a valid parent slot
+by u64::MAX.
+
+#### [next_slots](https://github.com/solana-labs/solana/blob/b57097ef18ab250aeb4ea8d3d111748ab4c41440/ledger/src/blockstore_meta.rs#L30-L32)
+next_slot is updated under [`handle_chaining`](https://github.com/yhchiang-sol/solana/blob/b57097ef18ab250aeb4ea8d3d111748ab4c41440/ledger/src/blockstore.rs#L933)
+inside [`chain_new_slot_to_prev_slot`](https://github.com/yhchiang-sol/solana/blob/b57097ef18ab250aeb4ea8d3d111748ab4c41440/ledger/src/blockstore.rs#L3543-L3550).
+As we never remove entries from the next_slots field, its merge operator can be
+implemented by taking the union of the new value and the existing value.
+
+#### [is_connected](https://github.com/solana-labs/solana/blob/b57097ef18ab250aeb4ea8d3d111748ab4c41440/ledger/src/blockstore_meta.rs#L33-L35)
+`is_connected` field is either updated based on
+[the previous slot](https://github.com/solana-labs/solana/blob/b57097ef18ab250aeb4ea8d3d111748ab4c41440/ledger/src/blockstore.rs#L3535)
+or via [propagation](https://github.com/solana-labs/solana/blob/b57097ef18ab250aeb4ea8d3d111748ab4c41440/ledger/src/blockstore.rs#L3469).
+As `is_connected` is never set from true to false, its merge operator can be
+implemented using the OR operation.
+
+#### [completed_data_indexes](https://github.com/solana-labs/solana/blob/b57097ef18ab250aeb4ea8d3d111748ab4c41440/ledger/src/blockstore_meta.rs#L36-L37)
+The `completed_data_indexes` remember the indexes which belonging slots
+are completed inside and are updated inside
+[`update_completed_data_indexes`](https://github.com/solana-labs/solana/blob/b57097ef18ab250aeb4ea8d3d111748ab4c41440/ledger/src/blockstore.rs#L3132-L3168).
+Its merge operator can simply burrow `update_completed_data_indexes`.
```

Comment:
burrow -> borrow

Reply:
Will the merge operator will have access to data from other column families as well? Needs the `ShredIndex` to figure out if all shreds in a range have been completed: https://github.com/solana-labs/solana/blob/b57097ef18ab250aeb4ea8d3d111748ab4c41440/ledger/src/blockstore.rs#L3164. Can we guarantee that this merge operation happens *after* the merge operation on the `ShredIndex` is done? Ordering seems important here